plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadataFactory.java

```diff
@@ -216,6 +219,7 @@ public HiveMetadataFactory(
         this.partitionProjectionEnabled = partitionProjectionEnabled;
         this.allowTableRename = allowTableRename;
         this.hiveViewsTimestampPrecision = requireNonNull(hiveViewsTimestampPrecision, "hiveViewsTimestampPrecision is null");
+        this.metadataFetchingExecutor = new BoundedExecutor(executorService, metadataParallelism);
```

Comment:
if parallelism is 1 then shouldn't we use `directExecutor` here to preserve old behaviour of "inline" metadata retreival. An executor with 1 thread != execution on calling thread.